libPhoneNumber/NBPhoneNumberUtil+ShortNumber.h

```diff
@@ -0,0 +1,198 @@
+ * {@link #isValidShortNumber} or {@link #isValidShortNumberForRegion}. Returns false if the
+ * number doesn't match the region provided.
+ *
+ * @param phoneNumber  the valid short number to check
+ * @param regionDialingFrom  the region from which the number is dialed
+ * @return  whether the short number is carrier-specific in the provided region, assuming the
+ *     input was a valid short number
+ */
+- (BOOL)isPhoneNumberCarrierSpecific:(NBPhoneNumber *)phoneNumber forRegion:(NSString *)regionCode;
+
+/**
+ * Given a valid short number, determines whether it is an SMS service (however, nothing is
+ * implied about its validity). An SMS service is where the primary or only intended usage is to
+ * receive and/or send text messages (SMSs). This includes MMS as MMS numbers downgrade to SMS if
+ * the other party isn't MMS-capable. If it is important that the number is valid, then its
+ * validity must first be checked using {@link #isValidShortNumber} or {@link
+ * #isValidShortNumberForRegion}. Returns false if the number doesn't match the region provided.
+ *
+ * @param phoneNumber  the valid short number to check
+ * @param regionDialingFrom  the region from which the number is dialed
+ * @return  whether the short number is an SMS service in the provided region, assuming the input
+ *     was a valid short number
+ */
+- (BOOL)isPhoneNumberSMSService:(NBPhoneNumber *)phoneNumber forRegion:(NSString *)regionCode;
+
+/**
+ * Returns true if the given number, exactly as dialed, might be used to connect to an emergency
+ * service in the given region.
+ * <p>
+ * This method accepts a string, rather than a PhoneNumber, because it needs to distinguish
+ * cases such as "+1 911" and "911", where the former may not connect to an emergency service in
+ * all cases but the latter would. This method takes into account cases where the number might
+ * contain formatting, or might have additional digits appended (when it is okay to do that in
+ * the specified region).
+ *
+ * @param number the phone number to test
+ * @param regionCode the region where the phone number is being dialed
+ * @return whether the number might be used to connect to an emergency service in the given region
+ */
+- (BOOL)connectsToEmergencyNumberFromString:(NSString *)number forRegion:(NSString *)regionCode;
+
+/**
+ * Returns true if the given number exactly matches an emergency service number in the given
+ * region.
+ * <p>
+ * This method takes into account cases where the number might contain formatting, but doesn't
+ * allow additional digits to be appended. Note that {@code isEmergencyNumber(number, region)}
+ * implies {@code connectsToEmergencyNumber(number, region)}.
+ *
+ * @param number the phone number to test
+ * @param regionCode the region where the phone number is being dialed
+ * @return whether the number exactly matches an emergency services number in the given region
+ */
+- (BOOL)isEmergencyNumber:(NSString *)number forRegion:(NSString *)regionCode;
```

Comment:
why are the arguments on these methods strings instead of NBPhoneNumber?

Reply:
According to the [canonical implementation](https://github.com/googlei18n/libphonenumber/blob/ed6b1ff4995ff562294aa00cdb718fa7888a0986/java/libphonenumber/src/com/google/i18n/phonenumbers/ShortNumberInfo.java#L431) comment, `isEmergencyNumber` implies `connectsToEmergencyNumber` and it requires a string because it needs to be able to distinguish between cases such as "+1 911" and "911", where some might connect and some might not.